src/main/java/frc/robot/Robot.java

```diff
@@ -17,39 +25,117 @@ public class Robot extends TimedRobot {
    * This function is run when the robot is first started up and should be used for any
    * initialization code.
    */
-  @Override
-  public void robotInit() {}
+
+  // drivetrain
+  private final WPI_VictorSPX frontLeft = new WPI_VictorSPX (1);
+  private final WPI_VictorSPX backLeft = new WPI_VictorSPX (2);
+  private final WPI_VictorSPX frontRight = new WPI_VictorSPX (3);
+  private final WPI_VictorSPX backRight = new WPI_VictorSPX (4);
+  private final MotorControllerGroup leftDrive = new MotorControllerGroup(backLeft, frontLeft);   
+  private final MotorControllerGroup rightDrive = new MotorControllerGroup(backRight, frontRight);
+  private final DifferentialDrive robotDrive = new DifferentialDrive(leftDrive, rightDrive);
+
+  // intake mechanism
+  private final CANSparkMax intake = new CANSparkMax(5, MotorType.kBrushless);
+  private final CANSparkMax rotation = new CANSparkMax(6, MotorType.kBrushless);
+
+
+  // joystick
+  private final Joystick driverJoystick = new Joystick(0);
+  private final Joystick operatorJoystick = new Joystick(1);
+  
+
+  public void robotInit() {
+    // drivetrain
+    rightDrive.setInverted(true);
+    frontLeft.setNeutralMode(NeutralMode.Brake);
+    backLeft.setNeutralMode(NeutralMode.Brake);
+    frontRight.setNeutralMode(NeutralMode.Brake);
+    backRight.setNeutralMode(NeutralMode.Brake);
+  }
+
 
   @Override
   public void robotPeriodic() {}
 
+
   @Override
   public void autonomousInit() {}
 
+
   @Override
   public void autonomousPeriodic() {}
 
+
   @Override
-  public void teleopInit() {}
+  public void teleopInit() {
+    // intializing speeds
+    robotDrive.arcadeDrive(0.0, 0.0);
+    intake.set(0.0);
+    rotation.set(0.0);
+  }
+
 
   @Override
-  public void teleopPeriodic() {}
+  public void teleopPeriodic() {
+    // arcadedrive command
+    robotDrive.arcadeDrive(-driverJoystick.getRawAxis(1), driverJoystick.getRawAxis(0));
+
+    //intake command
+    if (operatorJoystick.getRawButtonPressed(0)){
+      intake.set(0.4);
+    } 
+
+    if (operatorJoystick.getRawButtonReleased(0)) {
+      intake.set(0.0);
+    }
+
+    if (operatorJoystick.getRawButtonPressed(8)){
+      intake.set(-0.5);
+    }
+
+    if (operatorJoystick.getRawButtonReleased(8)) {
+      intake.set(0.0);
+    }
+    
+    if (operatorJoystick.getRawButtonPressed(5)){
+      rotation.set(0.3);
+    }
+
+    if (operatorJoystick.getRawButtonReleased(5)) {
+      rotation.set(0.0);
+    }
+
+    if (operatorJoystick.getRawButtonPressed(7)){
+      rotation.set(-0.3);
+    }
+
+    if (operatorJoystick.getRawButtonReleased(7)) {
+      rotation.set(0.0);
+    }
```

Comment:
This seems to be a lot of repeated code. I wonder if it would be possible to factor this out into a separate function? or multiple separate functions? perhaps one called `handleRotationButton()` and one called `handleIntakeButton()`. That way, `teleopPeriodic` remains clean and understandable.

Reply:
`public void teleopPeriodic() {
    // Drive command
    robotDrive.arcadeDrive(-driverJoystick.getRawAxis(1), driverJoystick.getRawAxis(0));
    
    // Intake commands
    handleIntakeButton(operatorJoystick);

    // Rotation commands
    handleRotationButton(operatorJoystick);
}

private void handleIntakeButton(Joystick joystick) {
    if (joystick.getRawButtonPressed(0)) {
        intake.set(0.4);
    } else if (joystick.getRawButtonReleased(0)) {
        intake.set(0.0);
    } else if (joystick.getRawButtonPressed(8)) {
        intake.set(-0.5);
    } else if (joystick.getRawButtonReleased(8)) {
        intake.set(0.0);
    }
}

private void handleRotationButton(Joystick joystick) {
    if (joystick.getRawButtonPressed(5)) {
        rotation.set(0.3);
    } else if (joystick.getRawButtonReleased(5)) {
        rotation.set(0.0);
    } else if (joystick.getRawButtonPressed(7)) {
        rotation.set(-0.3);
    } else if (joystick.getRawButtonReleased(7)) {
        rotation.set(0.0);
    }
}
`

Possible solution???

Reply:
Could be a way to format?

Reply:
i like it